xml/fingerprints.xsd

```diff
@@ -3,10 +3,101 @@
   <xsd:element name="fingerprints" type="fingerprints_element"/>
 
   <xsd:complexType name="fingerprints_element">
+    <xsd:annotation>
+      <xsd:documentation xml:lang="en">
+        The attributes that follow this documentation help categorize the
+        fingerprint databases and contents as well as enable optimization of
+        their processing.
+
+        - matches: provides a XML fingerprint database identifier. It is
+            intended to be descriptive in nature and often indicates a protocol
+            and function or sub-protocol type. Examples of this would be
+            'mysql.banners' vs 'mysql.errors'. If this attribute is not present
+            in the XML then its value is the base filename without the file
+            extension. See Recog::DB#parse_fingerprints
+            This value is returned as part of any successful match.
+
+        - protocol: indicates the application protocol of the fingerprints
```

Comment:
'application protocol' reads strange not just because I'm not sure that is a thing, but also because in this same description it is referred to as `service.protocol`.  Maybe reword this first sentence to "indicates the name of the service or protocol <something>".